### PR TITLE
[fmt] update to 8.0.1

### LINKED ISSUE #18577: [fmt] update to 8.0.0
Library name: fmt

New version number: 8.0.0

Other information that may be useful (release notes, etc...)
https://github.com/fmtlib/fmt/releases/tag/8.0.0

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #18577

Update fmt to the latest version 8.0.1

Note : no feature need to test

### CODE REVIEW

**Comment by serg06** [comment]
Here is one error from quill build:
```
D:\buildtrees\quill\src\v1.6.2-b72390cb62.clean\quill\include\quill/detail/misc/Os.h(151): error C2039: 'wmemory_buffer': is not a member of 'fmt'
```

And one from spdlog build:
```
D:\buildtrees\spdlog\src\v1.8.5-da0c3b7715.clean\include\spdlog/common-inl.h(63): error C2664: 'void fmt::v7::format_system_error(fmt::v7::detail::buffer<char> &,int,const char *) noexcept': cannot convert argument 3 from 'const std::string' to 'const char *'
```

Clearly `quill` and `spdlog` just don't work with fmt 8.0.0.

What needs to be done? Do we need to change `quill` and `spdlog`'s dependencies to say `"version<": "8.0.0"`?

**Comment by JonLiu1993** [comment]
> Here is one error from quill build:
> 
> ```
> D:\buildtrees\quill\src\v1.6.2-b72390cb62.clean\quill\include\quill/detail/misc/Os.h(151): error C2039: 'wmemory_buffer': is not a member of 'fmt'
> ```
> 
> And one from spdlog build:
> 
> ```
> D:\buildtrees\spdlog\src\v1.8.5-da0c3b7715.clean\include\spdlog/common-inl.h(63): error C2664: 'void fmt::v7::format_system_error(fmt::v7::detail::buffer<char> &,int,const char *) noexcept': cannot convert argument 3 from 'const std::string' to 'const char *'
> ```
> 
> Clearly `quill` and `spdlog` just don't work with fmt 8.0.0.
> 
> What needs to be done? Do we need to change `quill` and `spdlog`'s dependencies to say `"version<": "8.0.0"`?

We need to update the other two ports to the latest version

**Comment by serg06** [comment]
> We need to update the other two ports to the latest version

Just curious: Let's say the other two ports never get updated and never support fmt 8.0.0. What are our options?

**Comment by JonLiu1993** [comment]
> > We need to update the other two ports to the latest version
> 
> I'm confused. How can we update libraries that depend on fmt without updating fmt first?

For these interdependent libraries, we usually update them together, because after the local test is successful, the test on the CI will test the interdependence of the port.

**Comment by serg06** [comment]
@JonLiu1993 What happens if those libraries never get a new release, does that mean we can't update them thus we can't update fmt?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 What happens if those libraries never get a new release, does that mean we can't update them thus we can't update fmt?

I need to first determine if my update caused errors in the two ports that depend on it. If so, I need to modify it in fmt

**Comment by mathisloge** [comment]
https://github.com/gabime/spdlog/releases/tag/v1.9.0 got just a update for fmt::8

**Comment by mathisloge** [comment]
@JonLiu1993 https://github.com/odygrd/quill/releases/tag/v1.6.3 quill has now released. If updated, fmt update should be good

File: ports/fmt/portfile.cmake [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
 vcpkg_cmake_install()
-file(INSTALL ${SOURCE_PATH}/LICENSE.rst DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE.rst" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
 if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
     if(VCPKG_TARGET_IS_WINDOWS)
         if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
             if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/fmtd.dll")
-                file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/bin)
+                file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/bin")
                 file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/fmtd.dll ${CURRENT_PACKAGES_DIR}/debug/bin/fmtd.dll)
```
Comment (NancyLi1013) [inline ports/fmt/portfile.cmake:26]:
```suggestion
                file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/fmtd.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/fmtd.dll")
```

File: ports/fmt/portfile.cmake [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
             endif()
         endif()
         if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
             if(EXISTS "${CURRENT_PACKAGES_DIR}/lib/fmt.dll")
-                file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/bin)
+                file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/bin")
                 file(RENAME ${CURRENT_PACKAGES_DIR}/lib/fmt.dll ${CURRENT_PACKAGES_DIR}/bin/fmt.dll)
```
Comment (NancyLi1013) [inline ports/fmt/portfile.cmake:32]:
```suggestion
                file(RENAME "${CURRENT_PACKAGES_DIR}/lib/fmt.dll" "${CURRENT_PACKAGES_DIR}/bin/fmt.dll")
```

File: ports/fmt/vcpkg.json [hunk 3]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "fmt",
-  "version": "7.1.3",
-  "port-version": 5,
+  "version": "8.0.1",
```
Comment (NancyLi1013) [inline ports/fmt/vcpkg.json:3]:
```suggestion
  "version-semver": "8.0.1",
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you :D

**Pull request merged**